native/src/main/java/io/ballerina/stdlib/http/transport/contractimpl/sender/states/http2/ReceivingEntityBody.java

```diff
@@ -53,6 +59,7 @@ public class ReceivingEntityBody implements SenderState {
     private final Http2TargetHandler http2TargetHandler;
     private final Http2ClientChannel http2ClientChannel;
     private final Http2TargetHandler.Http2RequestWriter http2RequestWriter;
+    private Long contentLength = 0L;
```

Comment:
im not sure about storing the content length in the state itself. while receiving the entity body, there is a possibility for the state to change right? if it changes, when receiving the response body again, a new ReceivingEntityBody state will be created that doesn't have the previous contentLength value. This is a possibility right? @TharmiganK